pkg/k8s/watchers/cilium_network_policy.go

```diff
@@ -694,3 +683,31 @@ func cidrGroupRefsToCIDRsSets(cidrGroupRefs []string, cache map[string]*cilium_v
 	}
 	return cidrsSet, errors.Join(errs...)
 }
+
+func resetIngressIfEmpty(cnp *types.SlimCNP) {
+	if cnp.Spec != nil {
+		var found bool
+		for _, ingress := range cnp.Spec.Ingress {
+			if !ingress.IsNil() {
+				found = true
+				break
+			}
+		}
+		if !found {
+			cnp.Spec.Ingress = nil
+		}
+	}
+
+	for i := range cnp.Specs {
+		var found bool
+		for _, ingress := range cnp.Spec.Ingress {
+			if !ingress.IsNil() {
+				found = true
+				break
+			}
+		}
```

Comment:
This is incorrect. It's looping over `cnp.Spec.Ingress` and not `Specs`. I think this needs to be `cnp.Specs[i].Ingress`.

Reply:
And then ensure that this case is covered by the unit tests below as it doesn't seem to have been caught

Reply:
Godd catch, I'll add a test to check that, thanks!